### PR TITLE
Remove zero3 context manager from LoRA

### PR DESCRIPTION
**Pull request by Uxito-Ada**
### 1. Why the change?

#11167
This is an [unsolved issue](https://github.com/microsoft/DeepSpeed/issues/4703) in Deepspeed community, that zero3 context manager (which is needed by IPEX-LLM NF4 QLoRA) fails to find dist meta data e.g. world_size, and tensor position.

As LoRA BF16 is actually native huggingface transformers, zero3 context manager can be removed, which is only required by IPEX-LLM NF4 model.

### 2. User API changes

no

### 3. Summary of the change 

Remove zero3 context manager from LoRA

### 4. How to test?
- [ ] N/A
- [ ] Unit test
- [x] Application test
- [ ] Document test
- [ ] ...

### CODE REVIEW

File: python/llm/example/GPU/LLM-Finetuning/LoRA/alpaca_lora_finetuning.py
Comment (qiyuangong) [inline python/llm/example/GPU/LLM-Finetuning/LoRA/alpaca_lora_finetuning.py:1]: Do we need to remove `deepspeed_zero3` in parameter?

Comment (Uxito-Ada) [reply python/llm/example/GPU/LLM-Finetuning/LoRA/alpaca_lora_finetuning.py:1]: Yes, to judge whether it is needed to premove model to XPU.

**Review by qiyuangong — APPROVED** [review]
Other LGTM

**Review by plusbang — APPROVED** [review]
LGTM

**Pull request merged**